test-suite/js/src/test/scala/org/scalajs/testsuite/library/ObjectTest.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.library
+
+import org.junit.Assert._
+import org.junit.Test
+import org.scalajs.testsuite.utils.Platform
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.{ JSBracketAccess, JSName }
+
+class ObjectTest {
+  import ObjectTest._
+
+  @Test def getOwnPropertyDescriptors(): Unit = {
+    val obj1 = new js.Object {
+      val property1 = 42
+    }
+    val propertyDescriptors = js.Object.getOwnPropertyDescriptors(obj1)
+    val property = propertyDescriptors("property1")
+    assertTrue(property.writable)
+    assertEquals(property.value, 42)
+  }
+
+  /* These need to be lazy vals, so that they do not blow up if there is no
+   * symbol support at all.
+   */
+  private val symA: js.Symbol = js.Symbol("a")
+  private val symB: js.Symbol = js.Symbol.forKey("b")
+
+  @Test def getOwnPropertySymbols(): Unit = {
+    val obj = (new js.Object()).asInstanceOf[ObjectCreator]
+    obj(symA) = "localSymbol"
+    obj(symB) = "globalSymbol"
+    val objectSymbols = js.Object.getOwnPropertySymbols(obj)
+    assertEquals(objectSymbols.toSeq, Seq(symA, symB))
+  }
+
+  @Test def is(): Unit = {
+    val a = new js.Object()
+    assertTrue(js.Object.is(a, a))
+    assertTrue(js.Object.is(Double.NaN, Double.NaN))
+    assertTrue(js.Object.is(0.0, +0.0))
+    assertTrue(js.Object.is(-0.0, -0.0))
+    assertTrue(js.Object.is("foo", "foo"))
+
+    val b = new js.Object()
+    assertFalse(js.Object.is(a, b))
+    assertFalse(js.Object.is(-0.0, +0.0))
+  }
+
+  @Test def entries_from_object(): Unit = {
+    val obj = new js.Object {
+      val a = 42
+      val b = "foo"
+    }
+    val entries = js.Object.entries(obj)
+    val js.Tuple2(key1, value1) = entries(0)
+    assertEquals(key1, "a")
+    assertEquals(value1, 42)
+    val js.Tuple2(key2, value2) = entries(1)
+    assertEquals(key2, "b")
+    assertEquals(value2, "foo")
+  }
+
+  @Test def entries_from_dictionary(): Unit = {
+    val dict = js.Dictionary[Int]("a" -> 42, "b" -> 0)
+    val entries = js.Object.entries(dict)
+    val js.Tuple2(key1, value1) = entries(0)
+    assertEquals(key1, "a")
+    assertEquals(value1, 42)
```

Comment:
You should also add a test that makes sure the type of `value1` is correctly inferred as `Int`, which you can do with
```scala
val value1IsInt: Int = value1
assertEquals(42, value1IsInt)
```

Reply:
Also, please use the expected value on the left and the actual value on the right, in all tests.

Reply:
Fixed in https://github.com/scala-js/scala-js/pull/3869/commits/81c6972d8e07112d31e10356f5621e7c893e342a